### PR TITLE
Fix for assert during CSE processing

### PR DESCRIPTION
**Pull request by briansull**
Assertion failed '!comp->gtTreeHasSideEffects(tree, GTF_PERSISTENT_SIDE_EFFECTS_IN_CSE)'
Fixes VSO 569605

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x64 Checked jitstress1
@dotnet-bot test Windows_NT x64 Checked jitstress2
@dotnet-bot test Ubuntu x64 Checked jitstress1
@dotnet-bot test Ubuntu x64 Checked jitstress2
@dotnet-bot test Windows_NT x86 Checked jitstress1
@dotnet-bot test Windows_NT x86 Checked jitstress2

File: src/jit/optcse.cpp [hunk 1]
```diff
@@ -2515,18 +2514,16 @@ void Compiler::optValnumCSE_Heuristic()
...
+    Compiler::fgWalkResult result;
     assert(optValnumCSE_phase);
 
-    // If we have a non-empty *wbKeepList, then we first check for the rare case where we
-    // have a nested CSE def that has side-effects and return false if have this case
+    // First check for the rare case where we have a nested CSE def that has
+    // side-effects and return false whenever we have this case
     //
-    if (*wbKeepList != nullptr)
+    result = fgWalkTreePre(&deadTree, optHasCSEdefWithSideeffect, (void*)wbKeepList);
```
Comment (sandreenko) [inline src/jit/optcse.cpp]: I would like not to see `result` redefines and prefer smth like
```
Compiler::fgWalkResult defWithSideEffectResult = fgWalkTreePre(&deadTree, optHasCSEdefWithSideeffect, (void*)wbKeepList);
bool hasPersistentSideEffect = (defWithSideEffectResult == WALK_ABORT);
if (hasPersistentSideEffect)
```

**Comment by sandreenko** [comment]
cc @dotnet/jit-contrib

File: src/jit/optcse.cpp (hunk 1 above)
Comment (briansull) [reply src/jit/optcse.cpp]: OK, I updated it with your suggested changes

Comment (sandreenko) [reply src/jit/optcse.cpp]: Thank you, could you also delete `result` that became unused?

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x64 Checked jitstress1
@dotnet-bot test Windows_NT x64 Checked jitstress2
@dotnet-bot test Ubuntu x64 Checked jitstress1
@dotnet-bot test Ubuntu x64 Checked jitstress2
@dotnet-bot test Windows_NT x86 Checked jitstress1
@dotnet-bot test Windows_NT x86 Checked jitstress2

**Pull request merged**